packages/lesswrong/lib/collections/posts/schema.tsx

```diff
@@ -1016,6 +1017,34 @@ const schema: SchemaType<"Posts"> = {
     optional: true,
     hidden: true,
   },
+  
+  digestNum: resolverOnlyField({
+    graphQLtype: 'Int',
+    type: Number,
+    canRead: ['guests'],
+    resolver: async (post: DbPost, _, context: ResolverContext) => {
+      // only relevant to the EA Forum
+      if (!isEAForum) return null
+      return context.repos.posts.getDigestNumForPost(post._id)
+    },
+    // sqlResolver: ({field, join}) => join({
+    //   table: "DigestPosts",
+    //   type: "inner",
+    //   on: {
+    //     postId: field("_id"),
+    //     // onsiteDigestStatus: 'yes',
+    //   },
+    //   resolver: (digestPostsField) => join({
+    //     table: "Digests",
+    //     type: "inner",
+    //     on: {
+    //       _id: digestPostsField("digestId"),
+    //       // endDate: 'IS NOT NULL',
+    //     },
+    //     resolver: (digestsField) => `MAX(${digestsField("num")})`,
+    //   }),
+    // })
```

Comment:
I tried writing a `sqlResolver` but there seemed to be a few parts that didn't work, so I was concerned that I was going the wrong direction:
1. `onsiteDigestStatus: 'yes'` didn't work and I don't know why
2. `endDate: 'IS NOT NULL'` didn't work (I assume this was just not implemented?)
3. In order to do `MAX()` I need to add a `GROUP BY`

If my reviewer has thoughts on whether this is broadly correct and I can continue down this path (and how to fix #1) I'd appreciate them sharing. :) But it's pretty optional so I think it's fine to give up on this for now too.

Reply:
1. The values from `on` are directly interpolated into the SQL so you need to include the quotes for string literals: `"'yes'"`.
2. Yeah - this is a problem that isn't implemented yet.
3. Also not implemented yet due to complications around `GROUP BY`. I spoke to Will about this and we think it's probably safe to implement, but only if we only allow grouping on `p."_id"` which would be fine for this use case.
4. I think both these joins should probably have type `left` instead of `inner`.

It's uglier, but as a temporary work around you could just use a subquery instead:
```
    sqlResolver: ({field}) => `(
      SELECT MAX(d."num")
      FROM "Digests" d
      INNER JOIN "DigestPosts" dp ON
        dp."postId" = ${field("_id")} AND
        dp."digestId" = d."_id" AND
        dp."onsiteDigestStatus" = 'yes' AND
        d."endDate" IS NOT NULL
    )`,
```